### PR TITLE
TMDM-14722 Deploy datamodel fails after remove element from inheritance type

### PR DESCRIPTION
**Pull request by htyin**
**What is the current behavior?** (You should also link to an open issue here)

Redeploy failed when removing 0-many field from inheritance type

**What is the new behavior?**

Redeploy successfully when removing 0-many field from inheritance type.
Using declaring type to generate table name. 
Add Junit to test removing 0-many field from inheritance type

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
http://192.168.31.210:8080/view/Master/job/Master_03_TMDMEE_FOR-JIRA/498/
http://192.168.31.210:8080/view/Master/job/MDM-SOAP-EE-CentOS-MySQL8-DEBUG1/607/testReport/
http://192.168.31.210:8080/view/Master/job/MDM-REST-EE-MySQL8-CentOS-DEBUG3/131/HTML_20Report/

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LiquibaseSchemaAdapter.java [hunk 1]
```diff
@@ -130,6 +132,16 @@ protected String getTableName(FieldMetadata field) {
         }
         return tableName;
     }
+    
+    // Table is generated for 0-many simple field like 'entityName_x_fieldName'in database.
+    protected String getTableNameForField(FieldMetadata field) {
+        // For inheritance type, use declaringType to generate table name.       
+        String tableName = tableResolver.get((ComplexTypeMetadata) field.getDeclaringType()) + "_" + getColumnName(field);
```
Comment (hw-zhu) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LiquibaseSchemaAdapter.java:139]: add piece of code for (field.getDeclaringType() instanceof ComplexTypeMetadata) ;

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LiquibaseSchemaAdapter.java [hunk 2]
```diff
@@ -130,6 +132,16 @@ protected String getTableName(FieldMetadata field) {
...
+        if (dataSource.getDialectName() == DataSourceDialect.POSTGRES) {
+            tableName = tableName.toLowerCase();
+        }
+        return tableName;
```
Comment (hw-zhu) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LiquibaseSchemaAdapter.java:143]: In general, Database and table names are not case sensitive in Windows, and case sensitive in most varieties of Unix. so you'd better check the tableName for kinds of DB.

Comment (htyin) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LiquibaseSchemaAdapter.java:147]: We only do the check for postgres according to the existed logic to generate table name, 
 and test with other databases, no issue found.

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LiquibaseSchemaAdapter.java (hunk 1 above)
Comment (htyin) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LiquibaseSchemaAdapter.java]: ok

**Pull request merged**